WordPress/gutenberg
# Button Element: Add support to Site Editor

## What?
This adds support for button elements to the Site Editor.

## Why?
In https://github.com/WordPress/gutenberg/pull/40260 we added support for button elements, but it doesn't work in the Site Editor.

## How?
There is also a list of supported elements in the Site Editor, so we need to add buttons to this list.

## Testing Instructions
- Using this PR: https://github.com/Automattic/themes/pull/5849
- Open the Site Editor
- Add a button to a template
- Confirm that the button has the correct styles

## Screenshots or screencast 
<img width="1440" alt="Screenshot 2022-05-23 at 17 14 38" src="https://user-images.githubusercontent.com/275961/169862697-c5cd61cc-66f7-4a77-b674-baf0adc6449b.png">

@WordPress/block-themers
